### ISSUE TITLE
Cannot find module with root path using context

### ISSUE THREAD

**Issue by webdiscus** [webdiscus/pug-plugin#36]
`script(src=require('/js/main.js'))`:
- `3.0.1` - works
- `3.1.0` - doesn't work: `Cannot find module '/js/main.js'`

With styles is ok: `link(href=require('/styles/styles.scss'), rel='stylesheet')`

Structure:
- `src/views/index.pug`
- `src/js/main.js`
- `src/styles/styles.scss`

**Comment by webdiscus** [comment]
@silveoj

thanks for bug report. I will fix it.

**Comment by silveoj** [comment]
@webdiscus 
`v3.0.1`
Also I found one issue with styles:
I have created separated page to use it inside iframe. Inside this page I use again `link(href=require('/styles/styles.scss'), rel='stylesheet')`
I got `Can't resolve the file Images/image.png?format=jpg` for any images.

But I got the same error without iframe if I:
1. copy `styles.scss` and call it, e.g., `styles2.scss`
2. ```
    link(href=require('/styles/styles.scss'), rel='stylesheet')
    link(href=require('/styles/styles2.scss'), rel='stylesheet')```
3. The same error

**Comment by silveoj** [comment]
@webdiscus 
`v3.0.1`
Also I found one issue with styles:
I have created separated page to use it inside iframe. Inside this page I use again `link(href=require('/styles/styles.scss'), rel='stylesheet')`
I got `Can't resolve the file Images/image.png?format=jpg` for any images.

But I got the same error without iframe if I:
1. copy `styles.scss` and call it, e.g., `styles2.scss`
2. ```
    link(href=require('/styles/styles.scss'), rel='stylesheet')
    link(href=require('/styles/styles2.scss'), rel='stylesheet')```
3. The same error

I can try to prepare repo if need.

**Comment by webdiscus** [comment]
@silveoj 

Yes, please, create a repo. Thank You!

P.S. I have already fixed by me local the issuer with path relative by root context for js, like `script(src=require('/js/main.js'))`.
Today later will be the bugfix released.

**Comment by webdiscus** [comment]
@silveoj 

v3.1.1 with many bugfixes is released!
Perhaps this version fixes your problem with styles too. Please give me a feedback.

**Comment by silveoj** [comment]
@webdiscus Thanks.
js is ok
css in not ok. I will try prepare repo next 48h

**Comment by silveoj** [comment]
@webdiscus something is wrong with `postprocess` for js. I think `/main.js/` is not here in `postprocess` moment
```
new PugPlugin({
  postprocess: (content) => {
    // ...content
    content = content.replace(
      /src="(\/.*?\.js)"/g, // doesn't find anything
      `src="FROM(my_zip, '$1')"`,
    );
    return content;
  }),
})
```

`src="(\/.*?\.js)"` matches `src="/js/main.js"`

**Comment by webdiscus** [comment]
@silveoj 

`postprocess` due for HTML or CSS only.
JS will by bypassed in pug-plugin and never processed by pug-plugin.

**Comment by silveoj** [comment]
It's correct. But I replace my js and css links in HTML.

**Comment by silveoj** [comment]
It's correct. But I replace my js and css links in HTML.

Sometimes we need to paste HTML in another parser which read assets (images, css, js) from zip file and receive to client.

**Comment by webdiscus** [comment]
ok, I try to reproduce your case. You don't need create a repo.

**Comment by webdiscus** [comment]
The postprocess will be called after Pug compilation. At this stage the generated HTML contains already hashed assets output filenames, therefore you should match by output filename, not by source.

**Comment by webdiscus** [comment]
@silveoj please for new theme create NEW issue or discussion.

**Comment by webdiscus** [comment]
@silveoj 

I have already the test similarl to your case [resolve-context-image-pug-scss](https://github.com/webdiscus/pug-plugin/blob/master/test/cases/resolve-context-image-pug-scss/src/views/index.pug).
What is difference this test case in your usage case?

**Comment by silveoj** [comment]
@webdiscus  Also it doesn't work for js.
See `src\views\index.pug` in repo [css-js-duplicates branch](https://github.com/silveoj/pug-plugin-responsive-loader-issue/tree/css-js-duplicates)

### CSS

Your test case contains two classes with 2 different images. In example I use full copy of `styles.scss`.
`Can't resolve the file Images/image.png?size=200&format=webp`

### JS

Output for 2nd main.js is
 ```html
   <script src="C:\myProj\src\js\main.js"></script>
 ```
`Not allowed to load local resource: file:///C:/myProj/src/js/main.js`

**Comment by webdiscus** [comment]
@silveoj 

Exact this use case - many JS scripts with same source file is NOT allowed.
I know this limitation, see comment in the [test case](https://github.com/webdiscus/pug-plugin/blob/master/test/cases/resolve-context-script-ext/src/views/index.pug).

- this use case doesn't make sense. Do you have a practical use case that makes sense?
- to replace all double filenames require additional compilation time (even if there is only one file)

```pug
head
body 
  script(src=require('~Scripts/main.js'))
  //- second script with same file will not be replaced with output filename file:///C:/myProj/src/js/main.js
  //- script(src=require('~Scripts/main.js'))
```

**Comment by webdiscus** [comment]
@silveoj 
I can reproduce the bug with not resolved image. I will fix it.

**Comment by webdiscus** [comment]
@silveoj 

The issue with  responsive-loader will be fixed. This is the Bug. Thank You for bug report.

**Comment by webdiscus** [comment]
@silveoj 

to throw error while compilation require to detect a duplicate of script, what require additional compilation time.

**Comment by webdiscus** [comment]
@silveoj 

The issue is fixed in v3.1.2.
I have added experimental feature: display a warning when used duplicate scripts or styles in same Pug file.

Note: later, after tests and collection of statistics if the detection of duplicates take too much time (> 100 ms), it can be removed.

**Comment by silveoj** [comment]
@webdiscus It works.

But my situation is more interesting:
I see `[pug-plugin] WARNING  DON'T use duplicate scripts in same Pug!` message because of `iframe`

See the same repo and branch [css-js-duplicates branch](https://github.com/silveoj/pug-plugin-responsive-loader-issue/tree/css-js-duplicates)

Any chance to detect this situation in webpack?

**Comment by webdiscus** [comment]
@silveoj 

The bug is fixed in v3.1.3.

Test in your branch the following:
```pug
head
  link(rel="stylesheet", href=require('~Styles/styles.scss'))
  //- link(rel="stylesheet", href=require('~Styles/styles.scss'))

body 
  header Index2

  script(src=require('~Scripts/main.js'))
  //- script(src=require('~Scripts/main.js'))
```

Start in `webpack start:dev` and uncomment duplacates => display warnings.

**Comment by silveoj** [comment]
@webdiscus Thanks for quick result) Works as expected.
